simpleclient_servlet/src/main/java/io/prometheus/client/filter/MetricsFilter.java

```diff
@@ -0,0 +1,122 @@
+package io.prometheus.client.filter;
+
+import io.prometheus.client.Histogram;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+/**
+ * The MetricsFilter class exists to provide a high-level filter that enables tunable collection of metrics for Servlet
+ * performance.
+ *
+ * The Histogram name itself is required, and configured with a {@code metric-name} init parameter.
+ *
+ * By default, this filter will provide metrics that distinguish 3 levels deep for the request path
+ * (including servlet context path), but can be configured with the {@code path-components} init parameter.
+ *
+ * The Histogram buckets can be configured with a {@code buckets} init parameter whose value is a comma-separated list
+ * of valid {@code double} values.
+ *
+ * {@code
+ * <filter>
+ *   <filter-name>prometheusFilter</filter-name>
+ *   <filter-class>net.cccnext.ssp.portal.spring.filter.PrometheusMetricsFilter</filter-class>
+ *   <init-param>
+ *      <param-name>metric-name</param-name>
+ *      <param-value>webapp_metrics_filter</param-value>
+ *   </init-param>
+ *   <init-param>
+ *      <param-name>buckets</param-name>
+ *      <param-value>.001,.002,.005,.010,.020,.040,.080,.120,.200</param-value>
+ *   </init-param>
+ * </filter>
+ * }
+ *
+ * @author Andrew Stuart &lt;andrew.stuart2@gmail.com&gt;
+ */
+public class MetricsFilter implements Filter {
+    public static final String PATH_COMPONENT_PARAM = "path-components";
+    public static final String METRIC_NAME_PARAM = "metric-name";
+    public static final String BUCKET_CONFIG_PARAM = "buckets";
+
+    private Histogram servletLatency = null;
+
+    // Package-level for testing purposes
+    int pathComponents = 0;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        Histogram.Builder builder = Histogram.build()
+                .help("The time taken fulfilling servlet requests")
+                .labelNames("path", "method");
+
+        if (filterConfig == null) {
+            throw new ServletException("Please provide a configuration object, even for testing.");
+        }
+
+        if (StringUtils.isEmpty(filterConfig.getInitParameter(METRIC_NAME_PARAM))) {
```

Comment:
Please add a constructor that allows passing in any items used in the `FilterConfig`. This allows services using Spring Java Config to add a filter to be able to use compile time checks to provide useful values.